common/lib/plugin_service.ts

```diff
@@ -176,6 +175,31 @@ export class PluginService implements ErrorHandler, HostListProviderService {
     }
   }
 
+  async initiateTopologyUpdate(shouldVerifyWriter: boolean, timeoutMs: number): Promise<boolean> {
+    const hostListProvider = this.getHostListProvider();
+    if (!(hostListProvider instanceof MonitoringRdsHostListProvider)) {
```

Comment:
Instead of implementing this method that checks for a hard coded class (`hostListProvider instanceof MonitoringRdsHostListProvider`), I recommend implementing a new interface to keep `PluginService`'s implementation generic. JDBC uses this approach as well, see `BlockingHostListProvider`. I also recommend renaming this method to `forceMonitoringRefresh`